iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -1203,15 +1203,44 @@ static int subscribeToNotifyStateIfNeeded(PMQTTTRANSPORT_HANDLE_DATA transport_d
     return result;
 }
 
+// The tokenData variable carries a string in the format "<name>=<value".
+// This function returns the length of the <name> (the portion of the string from position 0 to the last char before '=').
+// If the '=' character is not present (unlikely), the length returned is 0, which should be
+// properly handled by the calling function.
+static int getPropertyLength(const char* tokenData)
```

Comment:
{
    int result = 0;

    while (tokenData[result] == '\0')
    {
        if (tokenData[result] == '=')
        {
            return result;
        }
        result++;
    }

    return 0;
}

Reply:
But what if tokenData does not have a terminating null character? Could there be a seg fault?

Reply:
This function has been removed. Unfortunately it didnt work for all the cases, according to how the MQTT transport is implemented. See comment on the issue discussion.